src/understand-tidb/plugin.md

```diff
@@ -1 +1,63 @@
 # Plugin
+
+The plugin API allows TiDB to be extended with new features such as audit logging or IP allow/deny listing.
+
+Sample code is provided for a basic audit logging plugin at [`./tidb/plugin/conn_ip_example/`](https://github.com/pingcap/tidb/tree/master/plugin/conn_ip_example). For an example on compiling TiDB and this plugin:
+
+```bash
+plugin="conn_ip_example"
+cd cmd/pluginpkg
+go install
+cd ../../plugin/$plugin
+pluginpkg -pkg-dir . -out-dir .
+cd ../..
+ 
+./bin/tidb-server -plugin-dir /path/to/tidb/plugin/$plugin -plugin-load $plugin-1
+```
+
+An explanation of what this does:
+
+- `cd cmd/pluginpkg` and `go install` compiles the command line utility called `pluginpkg`, which is used to build the plugin.
+- `pluginpkg -pkg-dir . -out-dir .` reads the plugin code + `manifest.toml` file and generates a shared object file for the plugin (`conn_ip_example-1.so`).
+- When the tidb-server starts, it can load plugins in a specified directory (`plugin-dir`).
+
+## Customizing the example plugin
+
+The manifest file describes the capabilities of the plugin, and which features it implements. For a basic version:
+
+```toml
+name = "conn_ip_example"
+kind = "Audit"
+description = "just a test"
+version = "1"
+license = ""
```

Comment:
```suggestion
license = "APLv2"
```

Reply:
As TiDB is licensed under APLv2 it should be compatible with APLv2 and GPLv3 at least (source: https://en.wikipedia.org/wiki/Apache_License#Compatibility ). I don't know if the way plugins are compiled and loaded allows the use of other licenses or not.

However I would like to see something here that would cause people to think about the license and change this to what's applicable for them. I also agree with what you said, so "APLv2" is probably not the right value here.

Maybe something like this:
```suggestion
license = "To be defined"
```

Or
```suggestion
license = "" # Please change this. See https://choosealicense.com/ for details
```

Or
```suggestion
license = "" # Suggested: APLv2 or GPLv3. See https://choosealicense.com/ for details
```

Reply:
I think recommending a license is worthwhile.  Can you create a PR on the original code here? https://github.com/pingcap/tidb/blob/d05660c834910c38a767fcdfc15c1b981863f499/plugin/conn_ip_example/manifest.toml#L5